### DISCUSSION TITLE
loading script pydiode.js locally

### DISCUSSION THREAD

**Discussion by Eliav2** [pyodide/pyodide#1990] [Q&A]
First of all - what a great project! 
I'm trying to load on an offline environment but can't make the script `pydiode.js` work locally the same way when called over URL.
```html
<head>
    <script src="https://cdn.jsdelivr.net/pyodide/v0.18.1/full/pyodide.js"></script> <!-- work -->
    <script src="src/pyodide.js"></script>                                          <!-- doesn't work -->
    <script src="./py-cra/client-hmr.js" type="module"></script>
</head>
```

why is that the case?

**Comment by rth** [comment]
If you open the JS console in your example you will see,
```
Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at 
    file:///.../pyodide.asm.js. (Reason: CORS request not HTTP).
```
This is the expected behavior due to browser security features. You would need to start a web-server to serve the files https://pyodide.org/en/stable/usage/downloading-and-deploying.html#serving-locally

**Reply by Eliav2** [reply]
Why is that? Can't I just load directly wasm or js module directly if it just lay in the src folder?
And actually, I don't get this cors error, for some reason I just can't load this file

**Reply by rth** [reply]
It's a security feature so that any webpage cannot open arbitrary files on your computer. cf https://stackoverflow.com/a/10752078/1791279

>  I don't get this cors error, for some reason I just can't load this file

What do you see in the "Console" tab Dev tools (F12 button)?

**Reply by rth** [reply]
BTW, there was a proposal to work around this security feature but it was fairly convoluted https://github.com/pyodide/pyodide/pull/606

**Reply by Eliav2** [reply]
I want to experiment with creating my own tiny python-based frontend framework, but I don't want to HTTP request from a remote server as this will not work offline and I don't want to force my potential users to run HTTP server to serve pyodide just to use this framework. 

here's a demo repo for quick reviews:  https://github.com/Eliav2/pydiode-test
you even don't have to clone: https://gitpod.io/#https://github.com/Eliav2/pydiode-test

I simply want it to get running offline without any additional webserver - is this possible?

**Reply by ryanking13** [reply]
You can also enable accessing local files on browser startup, which is not very recommended I guess.
https://stackoverflow.com/questions/4819060/allow-google-chrome-to-use-xmlhttprequest-to-load-a-url-from-a-local-file

**Reply by Eliav2** [reply]
@ryanking13 I'm looking for a solution so I could ship an experimental python framework without requiring the users to run another server, or disabling security features

**Reply by ryanking13** [reply]
For local development, I'm not sure but maybe you can serve files through bundler dev servers? In your case, Vite may have some options to provide files?

BTW, your code should not fail on `<script>` tag. CORS issue should happen at `loadPyodide` function call.
